code/controller/controller.h

```diff
@@ -1,32 +1,45 @@
 #ifndef CONTROLLER_H
 #define CONTROLLER_H
+#include "../utility/logger.h"
 
-enum Mode {
-    EMMA,
-    SAM,
-
-    NUM_MODES
-};
-
-enum Dir {
-    UP,
-    DOWN,
-    RIGHT,
-    LEFT,
-
-    NUM_DIRS
+struct Vector2i {
+	int x_point = 0;
+	int y_point = 0;
+	Vector2i(int xp, int yp) : x_point(xp), y_point(yp) {};
+	// TODO: Add ostream operator overloading
 };
 
 class Controller
 {
-
 public:
-    Controller(Mode);
-    void move(Dir);
-    void change_mode (Mode);
-    ~Controller();
-private:
-    Mode m_mode;
+    enum Mode {
+        EMMA,
+        SAM,
+
+        NUM_MODES
+    };
+
+    enum Dir {
+        UP,		// y > 0

```

Comment:
this is a github problem, because they use eight space tabs. Nothing I can do here unfortunately.
![tabs](https://cloud.githubusercontent.com/assets/7534496/19134603/ee379ca2-8b2c-11e6-89d9-0a8e0df9df4a.PNG)